woocommerce/woocommerce
# Added local pickup handling in `get_tax_location()`.

### All Submissions:






### Changes proposed in this Pull Request:
As described in #33135, When using the "Local Pickup" shipping method, taxes are calculated properly in the front-end checkout process, but if we **Recalculate** totals from the edit order screen in wp-admin, the tax didn't get calculated correctly.

For the "**Local Pickup**" shipping method, tax should be calculated based on store base address, but get_tax_location() is always return the tax location based on the value of the `woocommerce_tax_based_on` option of Tax settings without checking for local pickup shipping method. This PR contains changes to add a local pickup shipping method handing in the `get_tax_location()` function of the `WC_Abstract_Order` class, for consistent tax calculation behavior between the front-end and backend.

Closes #33135.

### How to test the changes in this Pull Request:
Detailed reproduce steps and screen recording can be found here https://github.com/woocommerce/woocommerce/issues/33135#issue-1242660490

1. Set store base address to somewhere in OH, USA (eg: 123, Some street, OH, USA, 43001).

2. Update Tax Options (**WooCommerce > Settings > Tax > Tax options**) as below.
![image](https://user-images.githubusercontent.com/10613171/169458188-79b07fa5-3f5d-4aff-9d9d-8a4325494e3b.png)

3. Set up a Tax Rate table for a single State, such as Ohio:
![image](https://user-images.githubusercontent.com/10613171/169458236-a59645ce-40fe-4cde-8322-d3fe3c2d9223.png)

4. Add "Shipping Zone" for the USA and add "Local Pickup" shipping method to Shipping Zone (taxable)

5. Attempt to purchase any product with an address that doesn't match the location in the tax rate table (i.e. the table is for US > OH, so use an address like Austin, TX 78701)

6. Select the "Local Pickup" method and notice the order total amount shown at checkout (which includes taxes), and place the order with the "Direct bank transfer" payment method.

7. Go to wp-admin and open the edit order screen of the order we just placed, Press `Recalculate`, and make sure the tax is calculated properly.

### Other information:




### FOR PR REVIEWER ONLY:
